src/shogun/kernel/MaternKernel.h

```diff
@@ -0,0 +1,140 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+
+#ifndef MATERNKERNEL_H
+#define MATERNKERNEL_H
+
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/kernel/ShiftInvariantKernel.h>
+
+namespace shogun
+{
+
+	/**
+	 * @brief The Matérn kernel
+	 *
+	 * It is defined as
+	 * \f[
+	 * k(x, y) = \frac{2 ^ {\nu - 1}}{\Gamma(\nu)}\Bigg(\sqrt{2\nu}\frac{D(x,
+	 *y)}{\rho}\Bigg)^\nu + N_{\nu}\Bigg(\sqrt{2\nu}\frac{D(x, y)}{\rho}\Bigg)
+	 * \f]
+	 * Where:
+	 * 		\f$N_{\nu}\f$ is the order of the modified Bessel function of the second kind
+	 *		\f$\rho\f$ is the kernel width
+	 *		\f$\Gamma\f$ is the gamma function
+	 **/
+	class MaternKernel : public ShiftInvariantKernel
+	{
+	public:
+		/** default constructor */
+		MaternKernel();
+
+		/** constructor
+		 *
+		 * @param width the kernel width
+		 * @param nu the order of the bessel function of the second kind
+		 * @param dist distance to be used
+		 */
+		MaternKernel(float64_t width, float64_t nu);
+
+		/** constructor
+		 *
+		 * @param size cache size
+		 * @param width the kernel width
+		 * @param nu the order of the bessel function of the second kind
+		 * @param dist distance to be used
+		 */
+		MaternKernel(int32_t size, float64_t width, float64_t nu);
```

Comment:
I'd suggest to remove this one ... the cache is useless for this kernel, so the size doesnt need to be set